src/test/java/org/takes/servlet/RqFromTest.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2014-2019 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.takes.servlet;
+
+import java.io.IOException;
+import org.cactoos.list.ListOf;
+import org.cactoos.text.JoinedText;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.StringContains;
+import org.hamcrest.core.StringStartsWith;
+import org.junit.jupiter.api.Test;
+import org.takes.rq.RqFake;
+import org.takes.rq.RqPrint;
+
+/**
+ * Test case for {@link  RqFrom}.
+ *
+ * @since 1.15
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
```

Comment:
@fabriciofx I think that you could create constants for the repeated values in this test and avoid using this supression WDYT?

Reply:
@paulodamaso Can I use the same constant to all tests? It will works only in this case...

Reply:
@fabriciofx Or you could use final variables inside your test methods, for example, 'final String get = "GET /a-test"' and `final String foo = "foo: bar"` for `header()` and use this variables in the test. You don't need to use constants because values are not shared between the tests.